packages/next/src/build/utils.ts

```diff
@@ -1713,7 +1713,7 @@ process.env.NODE_ENV = 'production'
 process.chdir(__dirname)
 
 const currentPort = parseInt(process.env.PORT, 10) || 3000
-const hostname = process.env.HOSTNAME || '0.0.0.0'
+const hostname = process.env.BIND_ADDRESS || process.env.HOSTNAME || '0.0.0.0'
```

Comment:
Lets add this env variable in the docs as well so users are aware of it https://github.com/ijjk/next.js/blob/canary/docs/01-app/05-api-reference/05-config/01-next-config-js/output.mdx#L54